mission7/queries.sql

```diff
@@ -0,0 +1,191 @@
+/*
+  1. 내 정보 업데이트 하기
+  - 닉네임을 "test"로 업데이트
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+UPDATE users
+SET nickname = 'test'
+WHERE id = 1;
+
+
+/*
+  2. 내가 생성한 상품 조회
+  - 현재 로그인한 유저 id가 1이라고 가정
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 3번째 페이지
+*/
+SELECT 
+  id, name, price, created_at
+FROM products
+WHERE user_id = 1
+ORDER BY created_at DESC
+LIMIT 10 OFFSET 20;
+
+
+/*
+  3. 내가 생성한 상품의 총 개수
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+SELECT COUNT(*) AS total_products
+FROM products
+WHERE user_id = 1;
+
+
+/*
+  4. 내가 좋아요 누른 상품 조회
+  - 현재 로그인한 유저 id가 1이라고 가정
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 3번째 페이지
+*/
+SELECT 
+  p.id, 
+  p.name, 
+  p.price, 
+  (SELECT i.image_url 
+     FROM images i 
+     WHERE i.product_id = p.id AND i.is_main = TRUE 
+     LIMIT 1) AS thumbnail,
+  l.created_at AS liked_at
+FROM likes l
+JOIN products p ON p.id = l.product_id
+WHERE l.user_id = 1
+ORDER BY l.created_at DESC
+LIMIT 10 OFFSET 20;
+
+
+/*
+  5. 내가 좋아요 누른 상품의 총 개수
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+SELECT COUNT(*) AS total_liked_products
+FROM likes
+WHERE user_id = 1 AND product_id IS NOT NULL;
+
+
+/*
+  6. 상품 생성
+  - 현재 로그인한 유저 id가 1이라고 가정
+  - 상품과 이미지 등록
+*/
+BEGIN;
+
+INSERT INTO products (user_id, name, description, price)
+VALUES (1, '테스트 상품', '테스트용 상품입니다.', 15000);
+
+INSERT INTO images (product_id, image_url, is_main)
+VALUES (
+  (SELECT id FROM products ORDER BY id DESC LIMIT 1),
+  'https://example.com/images/test-product.jpg',
+  TRUE
+);
+
+COMMIT;
```

Comment:
`SELECT id FROM products ORDER BY id DESC LIMIT 1`는 동시 트랜잭션에서 내가 방금 INSERT한 행이 아닌 다른 행의 id를 가져올 수 있습니다.
(READ COMMITTED에서 타 세션 커밋이 “가장 최신”으로 보이기 때문)

이미지가 잘못된 product_id에 연결되는 정합성 오류가 발생할 수 있어요.

한 트랜잭션 내에서, 추정(최댓값) 대신 DB가 반환하는 실제 PK를 사용하세요. 한 트랜잭션 내에서 경합이나 오염 없이 바로 바인딩합니다. 

```suggestion
BEGIN;

-- 1) 생성하고 id 받기
INSERT INTO products (user_id, name, description, price)
VALUES (1, '테스트 상품', '테스트용 상품입니다.', 15000)
RETURNING id;  -- 이 값을 사용

-- 2) 곧바로 이미지 삽입
INSERT INTO images (product_id, image_url, is_main)
VALUES (:product_id, 'https://example.com/images/test-product.jpg', TRUE);

COMMIT;
```